file_backed_device.go

```diff
@@ -73,25 +75,43 @@ type FileBackedDevice struct {
 type DriverConfig struct {
 	Source               *SyncSource
 	Backing              *SyncFile
-	NbdFileName          string
+	BlockDeviceName      string
 	ProcessFiles         []string
 	Fs                   FileSystem
 	Log                  *logrus.Logger
 	EnableBackgroundSync bool
 	Resumable            bool
+	UseDblockDriver      bool
 }
 
 // NewFileBackedDevice constructs a FileBackedDevice based on a source file
 func NewFileBackedDevice(
 	sourceFileName string,
 	backingFileName string,
-	nbdFileName string,
+	blockDeviceName string,
 	processFiles []string,
 	fs FileSystem,
 	log *logrus.Logger,
 	enableBackgroundSync bool,
 	resumable bool,
+	useDblockDriver bool,
 ) (*FileBackedDevice, error) {
+	if !useDblockDriver && !strings.HasPrefix(blockDeviceName, "/dev/nbd") {
+		return nil, fmt.Errorf("expected nbd block device name to begin with /dev/nbd*")
+	} else if useDblockDriver {
+		if !strings.HasPrefix(blockDeviceName, "/dev/") {
+			return nil, fmt.Errorf("block device must be placed in /dev/")
+		}
+
+		if _, err := os.Stat(blockDeviceName); !os.IsNotExist(err) {
+			return nil, fmt.Errorf("a block device with the name %s already exists", blockDeviceName)
+		}
+
+		if len(blockDeviceName) > (maxDiskNameLength + 5) {
+			return nil, fmt.Errorf("file name %s is too large", blockDeviceName)
```

Comment:
In this error message, how about providing the allowed max disk name length so an implementor does not necessarily need to refer back to the source?

Reply:
👍 did that change, thanks!